### PR TITLE
Add bindings for yarp/os/Log.h and yarp/os/LogStream.h

### PR DESCRIPTION
**Pull request by Tobias-Fischer**
Hi,
This adds bindings for yarp/os/Log.h and yarp/os/LogStream.h. It allows the following:
```
import yarp
logger = yarp.Log()
logger.debug('See how cool this is')
```

### CODE REVIEW

**Comment by towardthesea** [comment]
Tested the changes locally in my system without error!

File: bindings/yarp.i [hunk 1]
```diff
@@ -306,6 +306,8 @@ void setExternal2(yarp::sig::Image *img, PyObject* mem, int w, int h) {
 %enddef
 %define YARP_init_API
 %enddef
+%define YARP_OS_DEPRECATED_API
```
Comment (traversaro) [inline bindings/yarp.i]: Why is this line necessary?

Comment (Tobias-Fischer) [inline bindings/yarp.i]: The compilation fails otherwise with a "Swig input error" (or something like that). As far as I understand, Swig can't follow the #define of `YARP_OS_DEPRECATED_API` all the way down to where it is actually defined, as it is compiler dependent. I guess it's the same reason as for all the other `%define` lines above.

Comment (traversaro) [inline bindings/yarp.i]: Thanks.
